codeU1.py

```diff
@@ -76,43 +78,44 @@ def insert(self, value):
         new_node = Node(value)
         if not self.head:
             self.head = new_node;
+            self.tail = self.head;
             self.N = self.N+1;
-            return;
-        cur = self.head
-        while cur.next:
-            cur = cur.next   
-        cur.next = new_node 
-        self.N = self.N+1;
+            return;  
+        self.tail.next = new_node
+        self.tail = new_node
+        self.N+=1;
         
     def print_list(self):
         if not self.head:
-            print 'Empty list'
+            logging.warning('Empty list')
         cur = self.head
         while cur:
             print cur.val,
             cur = cur.next 
         print '\n'
-        return;    
+        return;  
+    
         
-#def k_last_element(x,k):
-#    print str(k) + 'th last element is ' + str(ll.k_last(k))     
-
-def test_list():      
-   x = [0,1,2,3]
-   ll = LinkedList();
-   for a in x:
-       ll.insert(a)
-   print 'List:'
-   ll.print_list()
-
-   print '0th last element is ' + str(ll.get_kth_last_element(0))
-   print '1th last element is ' + str(ll.get_kth_last_element(1))
-   print '2th last element is ' + str(ll.get_kth_last_element(4))
-
-
-print "="*25 + '\n Test singly-linked list \n' + "="*25
-test_list()
+class TestList(unittest.TestCase):
+    x = [0,1,2,3]
+    global ll
+    ll = LinkedList();
+    for a in x:
+        ll.insert(a) 
+    print '..Testing Single-Linked List'    
+    def test_insert(self):   
+        self.assertEqual(ll.N,4)
+        
+    def test_k_th_last_element(self):
+        self.assertEqual(ll.get_kth_last_element(0),3)
+        self.assertEqual(ll.get_kth_last_element(3),0)
+        try:
+            ll.get_kth_last_element(4)

```

Comment:
Should this one also be wrapped in `self.assertRaises` rather than in a `try` block?